Blockchain/task/src/blockchain/management/BlockChain.java

```diff
@@ -0,0 +1,114 @@
+package blockchain.management;
+
+import java.io.*;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static blockchain.utils.Constants.*;
+
+public class BlockChain implements Iterable<Block>, Serializable {
+    private Block head;
+    private Block tail;
+    private long size;
+    private AtomicLong lastBlockId = new AtomicLong(Long.parseLong(ZERO));
+    private AtomicInteger N = new AtomicInteger(Integer.parseInt(ZERO));
+    private NavigableSet<Transaction> transactions = new TreeSet<>(Comparator.comparingLong(Transaction::getId));
+    private ArrayList<MoneyHandler> moneyHandlers = new ArrayList<>();
+
+    public BlockChain() {}
```

Comment:
you don't need to specify constructor without any args , you have it be default, right?